src/openfhe.jl

```diff
@@ -381,3 +385,291 @@ function rotate(sv::SecureVector{<:OpenFHEBackend}, shift; wrap_by)
+    secure_matrix = SecureMatrix(ciphertext, size(sm1), capacity(sm1), sm1.context)
+
+    secure_matrix
+end
+
+function subtract(sm::SecureMatrix{<:OpenFHEBackend}, pm::PlainMatrix{<:OpenFHEBackend})
+    cc = get_crypto_context(sm)
+    ciphertext = OpenFHE.EvalSub(cc, sm.data, pm.data)
+    secure_matrix = SecureMatrix(ciphertext, size(sm), capacity(sm), sm.context)
+
+    secure_matrix
+end
+
+function subtract(pm::PlainMatrix{<:OpenFHEBackend}, sm::SecureMatrix{<:OpenFHEBackend})
+    cc = get_crypto_context(sm)
+    ciphertext = OpenFHE.EvalSub(cc, pm.data, sm.data)
+    secure_matrix = SecureMatrix(ciphertext, size(sm), capacity(sm), sm.context)
+
+    secure_matrix
+end
+
+function subtract(sm::SecureMatrix{<:OpenFHEBackend}, scalar::Real)
+    cc = get_crypto_context(sm)
+    ciphertext = OpenFHE.EvalSub(cc, sm.data, scalar)
+    secure_matrix = SecureMatrix(ciphertext, size(sm), capacity(sm), sm.context)
+
+    secure_matrix
+end
+
+function subtract(scalar::Real, sm::SecureMatrix{<:OpenFHEBackend})
+    cc = get_crypto_context(sm)
+    ciphertext = OpenFHE.EvalSub(cc, scalar, sm.data)
+    secure_matrix = SecureMatrix(ciphertext, size(sm), capacity(sm), sm.context)
+
+    secure_matrix
+end
+
+function negate(sm::SecureMatrix{<:OpenFHEBackend})
+    cc = get_crypto_context(sm)
+    ciphertext = OpenFHE.EvalNegate(cc, sm.data)
+    secure_matrix = SecureMatrix(ciphertext, size(sm), capacity(sm), sm.context)
+
+    secure_matrix
+end
+
+function multiply(sm1::SecureMatrix{<:OpenFHEBackend}, sm2::SecureMatrix{<:OpenFHEBackend})
+    cc = get_crypto_context(sm1)
+    ciphertext = OpenFHE.EvalMult(cc, sm1.data, sm2.data)
+    secure_matrix = SecureMatrix(ciphertext, size(sm1), capacity(sm1), sm1.context)
+
+    secure_matrix
+end
+
+function multiply(sm::SecureMatrix{<:OpenFHEBackend}, pm::PlainMatrix{<:OpenFHEBackend})
+    cc = get_crypto_context(sm)
+    ciphertext = OpenFHE.EvalMult(cc, sm.data, pm.data)
+    secure_matrix = SecureMatrix(ciphertext, size(sm), capacity(sm), sm.context)
+
+    secure_matrix
+end
+
+function multiply(sm::SecureMatrix{<:OpenFHEBackend}, scalar::Real)
+    cc = get_crypto_context(sm)
+    ciphertext = OpenFHE.EvalMult(cc, sm.data, scalar)
+    secure_matrix = SecureMatrix(ciphertext, size(sm), capacity(sm), sm.context)
+
+    secure_matrix
+end
+
+function rotate(sm::SecureMatrix{<:OpenFHEBackend}, shift)
+    # operate with data stored in matrix in form of vector
+    sv = SecureVector(sm.data, length(sm), sm.capacity, sm.context)
+    # minimum required shift
+    shift = shift .% size(sm)
+    # split algorithm in several cases depending on shift
+    if shift[1] == 0
+        shift_main = shift[2]*size(sm)[1]
+        sv = circshift(sv, shift_main; wrap_by=:length)
+    else
+        # mask for main part of single column
+        mask_part = zeros(size(sm)[1])
+        if shift[1] > 0
+            first = 1
+            last = size(sm)[1] - shift[1]
```

Comment:
Similarly, I think it makes the code easier to read if you use something descriptive like `nrows`/`ncols` instead of `size(sm)[i]`...